docs/prow/release-checklist.md

```diff
@@ -0,0 +1,89 @@
+# Release checklist
+
+This document describes how to execute release of Kyma using prow.
+
+## Release preparation
+
+This part can be done at any point of time. 
+This part applies only for new major and minor versions.
+
+1. Define release jobs in test-infra repository
+ - for every component
+ - for every tool
+ - kyma-integration
+ - kyma-gke-integration
+ - kyma-artifacts
+ - kyma-installer
+
+To see all release jobs for release 0.6, look for job names with prefix: `pre-rel06`.
+Every job name has to have unique name, so remember about prefixing it with `pre-rel<XY>`
+
+2. Ensure that tests for jobs exist
```

Comment:
At the same point should be something like: Add release branch name to `development/tools/jobs/tester/tester.go` file in `GetAllKymaReleaseBranches()` function.